scipy/integrate/_ivp/common.py

```diff
@@ -108,7 +114,10 @@ def select_initial_step(fun, t0, y0, f0, direction, order, rtol, atol):
         h0 = 1e-6
     else:
         h0 = 0.01 * d0 / d1
-
+    # Check t0+h0*direction doesn't take us beyond t_bound
+    h0 = min(h0, (t_bound-t0)/direction)
+    # Case where t0 = t_bound
+    if h0 == 0: return 0.
```

Comment:
Perhaps move this case right after `if y0.size == 0: ...` as a very degenerate.

Also probably better not to use one-liner ifs as they are not often used in the code base.